crates/tap-agent/src/test.rs

```diff
@@ -31,6 +58,184 @@ lazy_static! {
+        max_amount_willing_to_lose_grt: TRIGGER_VALUE + 100,
+        trigger_value: TRIGGER_VALUE,
+        rav_request_timeout: Duration::from_secs(30),
+        rav_request_receipt_limit: 1000,
+        indexer_address: INDEXER.1,
+        escrow_polling_interval: ESCROW_POLLING_INTERVAL,
+        tap_sender_timeout: Duration::from_secs(63),
+    }))
+}
+
+#[allow(clippy::too_many_arguments)]
+#[bon::builder]
+pub async fn create_sender_account(
+    pgpool: PgPool,
+    #[builder(default = HashSet::new())] initial_allocation: HashSet<Address>,
+    #[builder(default = TRIGGER_VALUE)] rav_request_trigger_value: u128,
+    #[builder(default = TRIGGER_VALUE)] max_amount_willing_to_lose_grt: u128,
+    escrow_subgraph_endpoint: Option<&str>,
+    network_subgraph_endpoint: Option<&str>,
+    #[builder(default = RECEIPT_LIMIT)] rav_request_receipt_limit: u64,
+    aggregator_endpoint: Option<Url>,
+) -> (
+    ActorRef<SenderAccountMessage>,
+    Arc<Notify>,
+    String,
+    Sender<EscrowAccounts>,
+) {
+    let config = Box::leak(Box::new(SenderAccountConfig {
+        rav_request_buffer: BUFFER_DURATION,
+        max_amount_willing_to_lose_grt,
+        trigger_value: rav_request_trigger_value,
+        rav_request_timeout: RAV_REQUEST_TIMEOUT,
+        rav_request_receipt_limit,
+        indexer_address: INDEXER.1,
+        escrow_polling_interval: Duration::default(),
+        tap_sender_timeout: TAP_SENDER_TIMEOUT,
+    }));
+
+    let network_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(network_subgraph_endpoint.unwrap_or(DUMMY_URL))
+                .unwrap(),
+        )
+        .await,
+    ));
+    let escrow_subgraph = Box::leak(Box::new(
+        SubgraphClient::new(
+            reqwest::Client::new(),
+            None,
+            DeploymentDetails::for_query_url(escrow_subgraph_endpoint.unwrap_or(DUMMY_URL))
+                .unwrap(),
+        )
+        .await,
+    ));
+    let (escrow_accounts_tx, escrow_accounts_rx) = watch::channel(EscrowAccounts::default());
+    escrow_accounts_tx
+        .send(EscrowAccounts::new(
+            HashMap::from([(SENDER.1, U256::from(ESCROW_VALUE))]),
+            HashMap::from([(SENDER.1, vec![SIGNER.1])]),
+        ))
+        .expect("Failed to update escrow_accounts channel");
+
+    let prefix = format!(
+        "test-{}",
+        PREFIX_ID.fetch_add(1, std::sync::atomic::Ordering::SeqCst)
+    );
+
+    let args = SenderAccountArgs {
+        config,
+        pgpool,
+        sender_id: SENDER.1,
+        escrow_accounts: escrow_accounts_rx,
+        indexer_allocations: watch::channel(initial_allocation).1,
+        escrow_subgraph,
+        network_subgraph,
+        domain_separator: TAP_EIP712_DOMAIN_SEPARATOR.clone(),
+        sender_aggregator_endpoint: aggregator_endpoint
+            .unwrap_or(Url::parse(&get_grpc_url().await).unwrap()),
```

Comment:
Could you do the same thing here? Create a match and call get_grpc_url() only in the None variant.